File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.java
class Solution {
    public int[] solution(int n, int s) {
        int[] answer = {};
        int tmp = s / n;
        if (tmp == 0) {
            answer = new int[] {-1};
        } else {
            answer = new int[n];
            for (int i = 0; i < n; i++) {
                answer[i] = tmp;
                if (i >= n - (s % n)) {
                    answer[i]++;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.py
def solution(n, s):
    answer = []
    tmp = s // n
    if tmp == 0:
        answer = [-1]
    else:
        answer = [tmp] * n
        if s % n > 0:
            for i in range(s%n):
                answer[n-1-i] += 1
    return answer